### ISSUE TITLE
FlutterBuild : Execution failed for task ':app:validateSigningRelease'

### ISSUE THREAD

**Issue by justoke** [FlorisDevreese/Flutter-CI-CD#1]
Hi,

I followed your article and have just got stuck with the signing of the release packages. I used the same build.gradle pattern you have which is to use environment variables. I tested the build on my PC with environment variables set and the resulting apk and aab was successfully signed. I was able to upload this to my Google Play developer console. However with the Azure Devops pipeline I have run into this problem. Any suggestions would be appreciated.

![image](https://user-images.githubusercontent.com/26506785/124096166-3ff31200-da52-11eb-8229-47f05fa40622.png)

![image](https://user-images.githubusercontent.com/26506785/124095795-eee31e00-da51-11eb-9f11-010b12960e46.png)

I've added a step to the pipeline to evaluate the variables and check that the KeyStore.jks file has been pulled from Secure Files.

![image](https://user-images.githubusercontent.com/26506785/124096307-5d27e080-da52-11eb-8ada-7cea4f2faafd.png)

My yml file, which is focused only on android at this stage.
`# Helper Pipeline
# Start with a minimal pipeline that you can customize to build and deploy your code.
# Add steps that build, run tests, deploy, and more:
# https://aka.ms/yaml

trigger:
- develop

pool:
  name: Azure Pipelines
  vmImage: windows-latest

steps:
  - task: DownloadSecureFile@1
    displayName: get KeyStore.jks
    name: releaseKeyStore
    inputs:
      secureFile: 'SignedKey-KeyStore.jks'
      retryCount: 8
      socketTimeout: 2000
  - task: DownloadSecureFile@1
    displayName: Get google-services.json
    name: googleServicesJson
    inputs:
      secureFile: 'google-services.json'
  - pwsh: Move-Item $(googleServicesJson.secureFilePath) android/app/ -Force
    displayName: Override google-services.json
  
  - pwsh: |
      Write-Host "##vso[task.setvariable variable=useEnvVariableStore;isOutput=true]$(useEnvVariableStore)"
      Write-Host "##vso[task.setvariable variable=keyAlias;isOutput=true]$(keyAlias)"
      Write-Host "##vso[task.setvariable variable=keyPassword;isOutput=true]$(keyPassword)"
      Write-Host "##vso[task.setvariable variable=storeFile;isOutput=true]$(releaseKeyStore.secureFilePath)"
      Write-Host "##vso[task.setvariable variable=storePassword;isOutput=true]$(keyPassword)"
    displayName: "Set keystore environment variables"

  - task: FlutterInstall@0
    displayName: "Flutter Install"
    inputs:
     channel: stable
     version: latest

  - script: |
      echo releaseKeyStore.secureFilePath : $(releaseKeyStore.secureFilePath)
      echo useEnvVariableStore : $(useEnvVariableStore)
      echo keyPassword : $(keyPassword)
      echo keyAlias : $(keyAlias)
      echo FlutterToolPath : $(FlutterToolPath)
      echo BuildNumber : $(buildNumber) 
      echo Package.Version : $(package.version)
      echo Show Sources Files: $(Build.SourcesDirectory)
      ls $(Build.SourcesDirectory)/*
      echo Show Release Files: $(Build.SourcesDirectory)/build/app/outputs/bundle/release/*
      ls $(Build.SourcesDirectory)/build/app/outputs/bundle/release/*
      echo Show Android App Files: 
      ls android/app/*
      cat android/app/build.gradle
      

    displayName: "Show variables and files"

  - task: FlutterBuild@0
    displayName: Build App
    inputs:
      target: 'apk'
      projectDirectory: '.'
      buildName: '$(Build.BuildNumber)'
      buildNumber: $(buildNumber)

  - task: PublishBuildArtifacts@1
    displayName: Publish artifact
    inputs:
      PathtoPublish: '$(Build.Repository.LocalPath)/build/app/outputs/apk/release'
      ArtifactName: 'apk'
      publishLocation: 'Container'`

**Comment by FlorisDevreese** [comment]
I've noted in the past that Azure DevOps has strange behavior with secret variables that have `*pass*` or `*password*` in their variable name.

It's like Azure DevOps does double encryption in the background, but only decrypts them only once...

Can you rename your variable name to avoid `*pass*` or `*password*`, and try again?

Please let me know your findings.
If this solves your issue, then I will report this to Microsoft.

**Comment by FlorisDevreese** [comment]
@justoke Any news here?

**Comment by justoke** [comment]
Hi. I managed to get this working finally. Seems the handling of secrets is fine and terms of them being correctly recovered in the pipeline when needed. However the issue was simply that I could not get environment variables to be loaded from the build.gradle script. They were always null. So I ended up using the File Creator from the marketplace and writing out the key store properties prior to the build. This works really well, just had to replace single backslash with double backslash when using a windows build agent.
